axa-footer: renders empty accordion on mobile

Hi,

MyAXA portal uses` "@axa-ch/footer": "3.0.8"` and I noticed bug in mobile design. 

We are using only one "column" section, not two as yours demo.
Empty accordion is rendered on mobile resolution.

I tried to reproduce it with the latest version, but it is not possible because of another issue #1832 .

Repo to reproduce (branch `myaxa-version`):
https://github.com/branislav-remen/patterns-lib-footer/tree/myaxa-version

![image](https://user-images.githubusercontent.com/43781862/86112523-5f1a3500-bac8-11ea-8eb2-bffbc04daacd.png)
![image](https://user-images.githubusercontent.com/43781862/86112643-8a9d1f80-bac8-11ea-9f59-561ddedf2bde.png)

Thanks.

I get the following error on my mac and following the "help steps" did not help. Weird that it works for you:

```
➜  patterns-lib-footer git:(myaxa-version) npm run start

> patterns-lib-footer@0.1.0 start /Users/c287512/workspace/patterns-lib-footer
> react-scripts start


There might be a problem with the project dependency tree.
It is likely not a bug in Create React App, but something you need to fix locally.

The react-scripts package provided by Create React App requires a dependency:

  "babel-eslint": "10.1.0"

Don't try to install it manually: your package manager does it automatically.
However, a different version of babel-eslint was detected higher up in the tree:

  /Users/c287512/workspace/node_modules/babel-eslint (version: 8.2.6) 

Manually installing incompatible versions is known to cause hard-to-debug issues.

If you would prefer to ignore this check, add SKIP_PREFLIGHT_CHECK=true to an .env file in your project.
That will permanently disable this message but you might encounter other issues.

To fix the dependency tree, try following the steps below in the exact order:

  1. Delete package-lock.json (not package.json!) and/or yarn.lock in your project folder.
  2. Delete node_modules in your project folder.
  3. Remove "babel-eslint" from dependencies and/or devDependencies in the package.json file in your project folder.
  4. Run npm install or yarn, depending on the package manager you use.

In most cases, this should be enough to fix the problem.
If this has not helped, there are a few other things you can try:

  5. If you used npm, install yarn (http://yarnpkg.com/) and repeat the above steps with it instead.
     This may help because npm has known issues with package hoisting which may get resolved in future versions.

  6. Check if /Users/c287512/workspace/node_modules/babel-eslint is outside your project directory.
     For example, you might have accidentally installed something in your home folder.

  7. Try running npm ls babel-eslint in your project folder.
     This will tell you which other package (apart from the expected react-scripts) installed babel-eslint.

If nothing else helps, add SKIP_PREFLIGHT_CHECK=true to an .env file in your project.
That would permanently disable this preflight check in case you want to proceed anyway.

P.S. We know this message is long but please read the steps above :-) We hope you find them helpful!

npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! patterns-lib-footer@0.1.0 start: `react-scripts start`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the patterns-lib-footer@0.1.0 start script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/c287512/.npm/_logs/2020-07-06T12_59_24_820Z-debug.log
```

But nevermind, I will create-react-app with your example and try to reproduce and otherwise (who knows) this dependency problem could be the issue. I will let you know. :)

--- a/src/components/30-organisms/footer/CHANGELOG.md
+++ b/src/components/30-organisms/footer/CHANGELOG.md
@@ -1,3 +1,7 @@
+## 4.1.2
+
+- Do not render an empty accordion on mobile if no content is given. #1836
+
 ## 4.1.0
 
 - Replaced old typography with new one. This changes could have changed the components design. (#1796 and #1750)

--- a/src/components/30-organisms/footer/CHANGELOG.md
+++ b/src/components/30-organisms/footer/CHANGELOG.md
@@ -1,6 +1,6 @@
 ## 4.1.2
 
-- Do not render an empty accordion on mobile if no content is given. #1836
+- Only render accordion on mobile when content is given. #1836
 
 ## 4.1.0
 

--- a/src/components/30-organisms/footer/demo.js
+++ b/src/components/30-organisms/footer/demo.js
@@ -63,4 +63,9 @@ storiesOf('Components|Footer/Demos', module)
     });
 
     return wrapper;
+  })
+  .add('Without content', () => {
+    const wrapper = document.createElement('div');
+    wrapper.innerHTML = `<axa-text>This is an emtpy footer. Just for testing.</axa-text><axa-footer></axa-footer>`;
+    return wrapper;
   });

--- a/src/components/30-organisms/footer/demo.js
+++ b/src/components/30-organisms/footer/demo.js
@@ -66,6 +66,6 @@ storiesOf('Components|Footer/Demos', module)
   })
   .add('Without content', () => {
     const wrapper = document.createElement('div');
-    wrapper.innerHTML = `<axa-text>This is an emtpy footer. Just for testing.</axa-text><axa-footer></axa-footer>`;
+    wrapper.innerHTML = `<axa-text>This demo is just for UI Testing. It shows the footer without content. Do not use it like this!</axa-text><axa-footer></axa-footer>`;
     return wrapper;
   });

--- a/src/components/30-organisms/footer/index.js
+++ b/src/components/30-organisms/footer/index.js
@@ -228,6 +228,7 @@ class AXAFooter extends InlineStyles {
           <div class="o-footer__content">
             <div class="o-footer__collection">
               <div class="o-footer__main">
+                ${this.querySelectorAll('[slot^="column-0-item-"]').length ? html`
                 <button
                   class="o-footer__accordion-button"
                   @click="${ev => this._handleAccordionClick(0, ev)}"
@@ -244,27 +245,30 @@ class AXAFooter extends InlineStyles {
                     ),
                     (item, index) => _renderFooterLinks(0, index)
                   )}
-                </ul>
+                </ul>` : ''}
               </div>
 
               <div class="o-footer__main">
-                <button
-                  class="o-footer__accordion-button"
-                  @click="${ev => this._handleAccordionClick(1, ev)}"
-                >
-                  <slot name="column-1-title" class="o-footer__title"></slot>
-                  <span class="${classMap(accordionCaretState(1))}">
-                    ${showCaret}
-                  </span>
-                </button>
-                <ul class="${classMap(shortAccordionContent)}">
-                  ${repeat(
-                    new Array(
-                      this.querySelectorAll('[slot^="column-1-item-"]').length
-                    ),
-                    (item, index) => _renderFooterLinks(1, index)
-                  )}
-                </ul>
+                ${this.querySelectorAll('[slot^="column-1-item-"]').length ? html`
+                  <button
+                    class="o-footer__accordion-button"
+                    @click="${ev => this._handleAccordionClick(1, ev)}"
+                  >
+                    <slot name="column-1-title" class="o-footer__title"></slot>
+                    <span class="${classMap(accordionCaretState(1))}">
+                      ${showCaret}
+                    </span>
+                  </button>
+                  <ul class="${classMap(shortAccordionContent)}">
+                    ${repeat(
+                      new Array(
+                        this.querySelectorAll('[slot^="column-1-item-"]').length
+                      ),
+                      (item, index) => _renderFooterLinks(1, index)
+                    )}
+                  </ul>
+                ` : ''}
+                
               </div>
             </div>
             <div class="o-footer__social-media">

--- a/src/components/30-organisms/footer/index.js
+++ b/src/components/30-organisms/footer/index.js
@@ -228,46 +228,65 @@ class AXAFooter extends InlineStyles {
           <div class="o-footer__content">
             <div class="o-footer__collection">
               <div class="o-footer__main">
-                ${this.querySelectorAll('[slot^="column-0-item-"]').length ? html`
-                <button
-                  class="o-footer__accordion-button"
-                  @click="${ev => this._handleAccordionClick(0, ev)}"
-                >
-                  <slot name="column-0-title" class="o-footer__title"></slot>
-                  <span class="${classMap(accordionCaretState(0))}">
-                    ${showCaret}
-                  </span>
-                </button>
-                <ul class="${classMap(accordionContent)}">
-                  ${repeat(
-                    new Array(
-                      this.querySelectorAll('[slot^="column-0-item-"]').length
-                    ),
-                    (item, index) => _renderFooterLinks(0, index)
-                  )}
-                </ul>` : ''}
+                ${
+                  this.querySelectorAll('[slot^="column-0-item-"]').length
+                    ? html`
+                        <button
+                          class="o-footer__accordion-button"
+                          @click="${ev => this._handleAccordionClick(0, ev)}"
+                        >
+                          <slot
+                            name="column-0-title"
+                            class="o-footer__title"
+                          ></slot>
+                          <span class="${classMap(accordionCaretState(0))}">
+                            ${showCaret}
+                          </span>
+                        </button>
+                        <ul class="${classMap(accordionContent)}">
+                          ${repeat(
+                            new Array(
+                              this.querySelectorAll(
+                                '[slot^="column-0-item-"]'
+                              ).length
+                            ),
+                            (item, index) => _renderFooterLinks(0, index)
+                          )}
+                        </ul>
+                      `
+                    : ''
+                }
               </div>
 
               <div class="o-footer__main">
-                ${this.querySelectorAll('[slot^="column-1-item-"]').length ? html`
-                  <button
-                    class="o-footer__accordion-button"
-                    @click="${ev => this._handleAccordionClick(1, ev)}"
-                  >
-                    <slot name="column-1-title" class="o-footer__title"></slot>
-                    <span class="${classMap(accordionCaretState(1))}">
-                      ${showCaret}
-                    </span>
-                  </button>
-                  <ul class="${classMap(shortAccordionContent)}">
-                    ${repeat(
-                      new Array(
-                        this.querySelectorAll('[slot^="column-1-item-"]').length
-                      ),
-                      (item, index) => _renderFooterLinks(1, index)
-                    )}
-                  </ul>
-                ` : ''}
+                ${
+                  this.querySelectorAll('[slot^="column-1-item-"]').length
+                    ? html`
+                        <button
+                          class="o-footer__accordion-button"
+                          @click="${ev => this._handleAccordionClick(1, ev)}"
+                        >
+                          <slot
+                            name="column-1-title"
+                            class="o-footer__title"
+                          ></slot>
+                          <span class="${classMap(accordionCaretState(1))}">
+                            ${showCaret}
+                          </span>
+                        </button>
+                        <ul class="${classMap(shortAccordionContent)}">
+                          ${repeat(
+                            new Array(
+                              this.querySelectorAll(
+                                '[slot^="column-1-item-"]'
+                              ).length
+                            ),
+                            (item, index) => _renderFooterLinks(1, index)
+                          )}
+                        </ul>
+                      `
+                    : ''
+                }
                 
               </div>
             </div>

--- a/src/components/30-organisms/footer/package.json
+++ b/src/components/30-organisms/footer/package.json
@@ -26,8 +26,8 @@
     "url": "https://github.com/axa-ch/patterns-library/issues"
   },
   "dependencies": {
-    "@axa-ch/container": "1.0.6",
-    "@axa-ch/materials": "^2.0.2",
+    "@axa-ch/container": "^3.0.3",
+    "@axa-ch/materials": "^10.1.0",
     "@skatejs/val": "^0.4.0",
     "lit-element": "^2.2.0",
     "lit-html": "^1.1.0"

--- a/src/components/30-organisms/footer/ui.test.js
+++ b/src/components/30-organisms/footer/ui.test.js
@@ -227,3 +227,25 @@ test('should render footer with working native callbacks', async t => {
 
   await t.expect($result.innerText).contains('https://www.facebook.com/axach/');
 });
+
+fixture('Footer - without content').page(
+  `${host}/iframe.html?id=components-footer-demos--without-content`
+);
+
+test('should not render empty accordions on mobile', async t => {
+  const $footerTitleColumn0 = Selector(() =>
+    document
+      .querySelector('axa-footer')
+      .shadowRoot.querySelector(`slot[name='column-0-title']`)
+  );
+  const $footerTitleColumn1 = Selector(() =>
+    document
+      .querySelector('axa-footer')
+      .shadowRoot.querySelector(`slot[name='column-1-title']`)
+  );
+
+  await t.expect($footerTitleColumn0.exists).eql(false);
+  await t.expect($footerTitleColumn1.exists).eql(false);
+}).before(async t => {
+  await t.resizeWindow(767, 767);
+});

